### PR TITLE
[DUOS-2949] Get by ID DAA API

### PR DESCRIPTION
**Pull request by aarohinadkarni**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2949

### Summary
Adds a new endpoint to get a Data Access Agreement by its ID along with API documentation

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/assets/api-docs.yaml [hunk 1]
```diff
@@ -122,6 +122,8 @@ paths:
       responses:
         404:
           description: Deprecated endpoint. Use  `GET /api/tdr/{identifier}` instead.
+  /api/daa/{daaId}:
+    $ref: 'paths/daaByDaaId.yaml'
   /api/daa/{dacId}:
     $ref: 'paths/daaById.yaml'
```
Comment (rushtong) [inline src/main/resources/assets/api-docs.yaml]: We can't actually do this ... we might need a refactor of `/api/daa/{dacId}:` to be something like `/api/daa/dac/{dacId}:` or something similar. The two paths (as defined) have the same signature (`/api/daa/<integer>`) so Jersey cannot distinguish between the two.

Comment (aarohinadkarni) [reply src/main/resources/assets/api-docs.yaml]: For now, I've refactored the POST api signature.

File: src/main/java/org/broadinstitute/consent/http/service/DaaService.java [hunk 2]
```diff
@@ -84,4 +86,11 @@ public void removeDacFromDaa(Integer dacId, Integer daaId) {
     daaDAO.deleteDacDaaRelation(dacId, daaId);
   }
 
+  public DataAccessAgreement findById(Integer daaId) {
+    DataAccessAgreement daa = daaDAO.findById(daaId);
+    if (Objects.nonNull(daa)) {
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DaaService.java]: nit:
```suggestion
    if (daa != null) {
```

**Review by rushtong — APPROVED** [review]
One minor suggestion inline, but 👍🏽 overall!

**Review by fboulnois — APPROVED** [review]
looks good 👍

**Pull request merged**